lib/jekyll-import/importers/dotclear.rb

```diff
@@ -1,121 +1,101 @@
 # frozen_string_literal: true
 
-# Tested with dotClear 2.1.5
 module JekyllImport
   module Importers
     class Dotclear < Importer
-      def self.specify_options(c)
-        c.option "datafile", "--datafile PATH", "dotClear export file"
-        c.option "mediafolder", "--mediafolder PATH", "dotClear media export folder (media.zip inflated)"
-      end
-
-      def self.require_deps
-        JekyllImport.require_with_fallback(%w(
-          rubygems
-          fileutils
-          safe_yaml
-          date
-          active_support
-          active_support/core_ext/string/inflections
-          csv
-          pp
-        ))
-      end
-
-      def self.validate(opts)
-        abort "Specify a data file !" if opts["datafile"].nil? || opts["datafile"].empty?
-        abort "Specify a media folder !" if opts["mediafolder"].nil? || opts["mediafolder"].empty?
-      end
-
-      def self.extract_headers_section(str)
-        str[1..-2].split(" ")[1].split(",")
-      end
-
-      def self.extract_data_section(str)
-        str.gsub(%r!^"!, "").gsub(%r!"$!, "").split('","')
-      end
-
-      def self.process(opts)
-        options = {
-          :datafile    => opts.fetch("datafile", ""),
-          :mediafolder => opts.fetch("mediafolder", ""),
-        }
-
-        FileUtils.mkdir_p("_posts")
-        FileUtils.mkdir_p("_drafts")
-
-        type_data = ""
-        headers = {}
-        posts_and_drafts = {}
-        keywords = {}
-
-        File.readlines(options[:datafile]).each do |lineraw|
-          line = lineraw.strip.gsub(%r!\n$!, "")
+      class << self
+        def specify_options(c)
+          c.option "datafile", "--datafile PATH", "Dotclear export file"
+        end
 
-          next if line.empty?
+        def require_deps
+          JekyllImport.require_with_fallback(%w(
+            csv
+            reverse_markdown
+            yaml
+          ))
+        end
 
-          if line.start_with?("[") # post | media \ meta | comment...
-            type_data = line.split(" ").first[1..-1]
-            headers[type_data] = extract_headers_section(line)
-            next
+        # rubocop:disable Style/IfUnlessModifier
+        def validate(opts)
+          file_path = opts["datafile"]
+          if file_path.nil? || file_path.empty?
+            Jekyll.logger.abort_with "Import Error:", "Dotclear export file not found!"
```

Comment:
Should we be clear that this comes from the "datafile" option?

Reply:
That sounds better! 👍